domain/src/main/java/org/sagebionetworks/research/domain/step/ui/action/ActionHandler.java

```diff
@@ -30,14 +30,23 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-package org.sagebionetworks.research.domain.step.ui.action.interfaces;
+package org.sagebionetworks.research.domain.step.ui.action;
 
-import org.sagebionetworks.research.domain.step.interfaces.Step;
+import android.support.annotation.NonNull;
 
-import java.util.Optional;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
 
 public interface ActionHandler {
-    Optional<Action> getAction(ActionType type, Step step);
```

Comment:
I'm wondering if we should have an ActionViewHandler in the presentation layer rather than having this in the domain. This way the Action class we will be using will make use of DisplayString.

Reply:
Yea, we'll need to have something in the presentation layer. This was a refactor to match iOS, and moved getActions from the UIStep to this interface. I think we already have something in the presentation layer that handles this for steps (but we still need to do this for the task)

Reply:
Yea, we'll need to have something in the presentation layer. This was a refactor to match iOS, and moved getActions from the UIStep to this interface. I think we already have something in the presentation layer that handles this for steps (but we still need to do this for the task)